engine_test.go

```diff
@@ -706,48 +706,93 @@ func TestStarPanic197(t *testing.T) {
 }
 
 func TestIndexes(t *testing.T) {
-	require := require.New(t)
 	e := newEngine(t)
 
 	tmpDir, err := ioutil.TempDir(os.TempDir(), "pilosa-test")
-	require.NoError(err)
+	require.NoError(t, err)
 
-	require.NoError(os.MkdirAll(tmpDir, 0644))
+	require.NoError(t, os.MkdirAll(tmpDir, 0644))
 	e.Catalog.RegisterIndexDriver(pilosa.NewIndexDriver(tmpDir))
 
 	db, err := e.Catalog.Database("mydb")
-	require.NoError(err)
+	require.NoError(t, err)
 	table := db.Tables()["mytable"].(sql.Indexable)
 
 	driver := e.Catalog.IndexDriver(pilosa.DriverID)
 	conf := make(map[string]string)
 	expr := sql.NewExpressionHash(expression.NewGetFieldWithTable(0, sql.Int64, "mytable", "i", false))
 	idx, err := driver.Create("mydb", "mytable", "myidx", []sql.ExpressionHash{expr}, conf)
-	require.NoError(err)
+	require.NoError(t, err)
 
 	created, err := e.Catalog.AddIndex(idx)
-	require.NoError(err)
+	require.NoError(t, err)
 
 	iter, err := table.IndexKeyValueIter(sql.NewEmptyContext(), []string{"i"})
-	require.NoError(err)
+	require.NoError(t, err)
 
-	require.NoError(driver.Save(sql.NewEmptyContext(), idx, iter))
+	require.NoError(t, driver.Save(sql.NewEmptyContext(), idx, iter))
 	created <- struct{}{}
 
 	defer func() {
 		done, err := e.Catalog.DeleteIndex("foo", "myidx", true)
-		require.NoError(err)
+		require.NoError(t, err)
 		<-done
 	}()
 
-	_, it, err := e.Query(sql.NewEmptyContext(), "SELECT * FROM mytable WHERE i = 2")
-	require.NoError(err)
+	testCases := []struct {
```

Comment:
shall we test indexes made from more than one column too?

Reply:
and some AND and OR

Reply:
We cannot test `AND` and `OR` because there is no `SetOperations` implemented yet in pilosa

Reply:
Added the tests for more than one column